src/control.rs

```diff
@@ -38,8 +36,14 @@ pub fn set_virtual_terminal(use_virtual: bool) -> Result<(), ()> {
 
     unsafe {
         let handle = GetStdHandle(STD_OUTPUT_HANDLE);
+        if handle == INVALID_HANDLE_VALUE {
+            return Err(());
+        }
+
         let mut original_mode: DWORD = 0;
-        GetConsoleMode(handle, &mut original_mode);
+        if GetConsoleMode(handle, &mut original_mode) == 0 {
```

Comment:
Do you know if we can replace the `0` by a constant with a meaning ?

Reply:
No, [the docs](https://learn.microsoft.com/en-us/windows/console/getconsolemode#return-value) literally just say:
> If the function fails, the return value is zero.

Reply:
Ok, thank you for that.

Could you add either a comment with this URL or something like this ?

```suggestion
        const IS_ERROR = 0;
        if GetConsoleMode(handle, &mut original_mode) == IS_ERROR {
```